src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1519,4 +1519,16 @@
     <value>and all other platforms</value>
     <comment>This call site is reachable on: 'windows' 10.0.2000 and later, and all other platforms</comment>
   </data>
+  <data name="DoNotGuardDictionaryRemoveByContainsKeyDescription" xml:space="preserve">
+    <value>Do not guard 'Dictionary.Remove(key)' with 'Dictionary.ContainsKey(key)'. 'Remove()' already checks whether the key exists, and will not throw if it does not.</value>
+  </data>
+  <data name="DoNotGuardDictionaryRemoveByContainsKeyMessage" xml:space="preserve">
+    <value>Do not guard 'Dictionary.Remove(key)' with 'Dictionary.ContainsKey(key)'</value>
+  </data>
+  <data name="DoNotGuardDictionaryRemoveByContainsKeyTitle" xml:space="preserve">
+    <value>Do not guard 'Dictionary.Remove(key)' with 'Dictionary.ContainsKey(key)'</value>
```

Comment:
I am wondering if we would want to the meaning to be more on the meaning "Redundant" rather than "Do not" .

Reply:
OK, this I'm unsure about. I see how "Do not" is a bit harsh (suggesting a problem, rather than a suggestion), but I can't find a good alternative in the existing resources. (Some are prefixed "Avoid", but that sounds even harsher to me.)

(And most start with a verb, so "redundant" wouldn't work.)

Reply:
That's fine, sometimes I just write out what I think, doesn't mean actions have to be made.

@mavasani It could be an interesting subject to think more globally about messages to make them informative and supportive while not being too harsh.

Reply:
How about keeping it small and simple: `Unnecessary call to 'Dictionary.ContainsKey'`? @buyaa-n?